### ISSUE TITLE
installation

### ISSUE THREAD

**Issue by mebrooks** [DTUAqua/hpc_setup#3]
While I was troubleshooting my other issue, I started to suspect that the setup never worked (no evidence that `save(x, file="x.Rdata")` was run at end of a test script). So I tried installing again and I get this error.

```
checking if libcurl is version 7 and >= 7.22.0... no
configure: error: libcurl >= 7.22.0 library and headers are required with support for https
make[1]: Entering directory `/zhome/61/6/119429/hpc_setup/build'
make[1]: *** No targets specified and no makefile found.  Stop.
make[1]: Leaving directory `/zhome/61/6/119429/hpc_setup/build'
```
I can also see that my `Makevars` file is empty.

Any ideas on how to troubleshoot this?

**Comment by kaskr** [comment]
Are you using the new login host `ssh $USER@login2.gbar.dtu.dk` ?

https://github.com/DTUAqua/hpc_setup#get-started

**Comment by mebrooks** [comment]
I was using the old login host `login`, but I get the same result with the new one. I even tried deleting everything and starting over.

**Comment by kaskr** [comment]
I get:

```shell
n-62-29-1(kaskr) $ curl-config --version
libcurl 7.29.0
```

And R picks up the right version:

```
checking for curl-config... /usr/bin/curl-config
checking libcurl version ... 7.29.0
checking curl/curl.h usability... yes
checking curl/curl.h presence... yes
checking for curl/curl.h... yes
checking if libcurl is version 7 and >= 7.22.0... yes
checking if libcurl supports https... yes
```

I wonder if you have some old junk in your PATHs ?

**Comment by mebrooks** [comment]
Yes, it could be the paths. How do I check that? 

I get 
```
n-62-29-1(molbr) $ curl-config --version
libcurl 7.63.0
```

**Comment by kaskr** [comment]
But wait... Your version is even newer than mine. It makes no sense you get 
`checking if libcurl is version 7 and >= 7.22.0... no`
You must have been on an old login node when you did the R install. Perhaps try R install again **after** verifying the new curl version.

**Comment by mebrooks** [comment]
How do I install R? #1

**Comment by mebrooks** [comment]
I have this in my `.bashrc` file
```
  export PATH=/appl/R/3.5.2-sl64/bin:$PATH
```
but sometimes I need to use `R-3.5.3-mkl2019` because otherwise I got errors when installing packages as below
```
> install.packages(c("knitr", "rmarkdown", "ggplot2", "bbmle", "plyr"), type="source")
Installing packages into '/zhome/61/6/119429/R/x86_64-pc-linux-gnu-library/3.5'
(as 'lib' is unspecified)
Warning: unable to access index for repository https://mirrors.dotsrc.org/cran/src/contrib:
  internet routines cannot be loaded

> remotes::install_github("mebrooks/selfisher/selfisher")
Error in utils::download.file(url, path, method = download_method(), quiet = quiet,  : 
  internet routines cannot be loaded
```

**Comment by kaskr** [comment]
That's a problem: `export PATH=/appl/R/3.5.2-sl64/bin:$PATH` no longer needed.
Please fix your `.bashrc`(or whatever) and remove `hpc_setup` and clone it from scratch. It uses version 3.6.2 which is newer, so no problem with package updates.

**Comment by mebrooks** [comment]
Should I also delete 
```
module load gcc/6.1.0
```
?

**Comment by kaskr** [comment]
Yes

**Comment by mebrooks** [comment]
I was making a lot of progress with installation. The makefile had a hangup with installing the `devtools` package and I had to go into R and install a bunch of packages, sometimes looking for errors like 
```
Error in dyn.load(file, DLLpath = DLLpath, ...) : 
  unable to load shared object '/zhome/61/6/119429/R/x86_64-pc-linux-gnu-library/3.6/foo/libs/foo.so':
  libR.so: cannot open shared object file: No such file or directory
```
and then I would `install.packages("foo")` for whatever `foo` was in the error message.

After I got `devtools` installed, I ran `MAKE="make -j8" make all` again. Then I got stuck at a point where it couldn't install the `V8` package. This is the error message
```
Using PKG_LIBS=-lv8 -lv8_libplatform
-----------------------------[ ANTICONF ]-------------------------------
Configuration failed to find the libv8 engine library. Try installing:
 * deb: libv8-dev or libnode-dev (Debian / Ubuntu)
 * rpm: v8-devel (Fedora, EPEL)
 * brew: v8 (OSX)
 * csw: libv8_dev (Solaris)
To use a custom libv8, set INCLUDE_DIR and LIB_DIR manually via:
R CMD INSTALL --configure-vars='INCLUDE_DIR=... LIB_DIR=...'
---------------------------[ ERROR MESSAGE ]----------------------------
<stdin>:1:10: fatal error: v8.h: No such file or directory
compilation terminated.
------------------------------------------------------------------------
ERROR: configuration failed for package 'V8'
* removing '/zhome/61/6/119429/R/x86_64-pc-linux-gnu-library/3.6/V8'

```

**Comment by mebrooks** [comment]
The HPC maintainers installed `v8` to solve that problem. Then, I ran `MAKE="make -j8" make all` again. I went through several of the same rounds as above with messages about `.so` files and solved them the same as above. I don't know what's wrong with my setup that it wasn't automatically working. Maybe it was because I already had things installed and complied under different settings. Anyhow, in the end I got it to work. It just took time and patience.
